website/guide/programming.html

```diff
@@ -493,6 +493,28 @@ <h2 id="duktape-api">Duktape API</h2>
 duk_get_prop_string(ctx, -1, "my_state_variable");
 </pre>
 
+
+<h3 id="hidden-symbol-properties">Hidden Symbol properties</h3>
+
+<p>If data needs to be associated with an object, but hidden from Ecmascript
+code, hidden <a href="#symbols">Symbols</a> can be used as a property key. The
+key is only accessible via the C API, unless passed to Ecmascript code from the
+C API. A common use case is to associated backing pointers/data to C memory.
+Symbols are created as a string, but are differentiated with macros that mark
+them as Symbols.</p>
+
+<p>For example, setting and getting a hidden symbol on <code>this</code>:</p>
+<pre class="c-code">
+my_context_data_t *my_context_data = malloc(sizeof(my_context_data_t));
+duk_push_this(ctx);
+duk_push_pointer(ctx, my_context_data);
+duk_put_prop_string(ctx, -2, DUK_HIDDEN_SYMBOL("my_context_data"));
+/* ... */
+duk_push_this(ctx);
+duk_get_prop_string(ctx, -1, DUK_HIDDEN_SYMBOL("my_context_data"));
+my_context_data_t *my_context_data = duk_get_pointer(ctx, -1);
+</pre>
+
```

Comment:
Placing the section here is a bit of a compromise: the H2 heading talks about function state. Then again, the H2 level section talks about stashes etc which are general concepts and not restricted to function state. (The guide is in need of some rewriting in parts, and the programming model section is one of those parts ... :)

But overall, I think this is a good enough location for now.